app/src/main/java/com/example/bottomnav/Parse.java

```diff
@@ -51,75 +51,79 @@ public Parse(char[] code) throws Exception {
     }
 
     private void parseComments(IASTNode[] comments) {
-        Pattern pattern1 = Pattern.compile("@canPayloadStruct\\s+(\\S+)\\s*=\\s*(\\S+)");
+        Pattern pattern = Pattern.compile("@canPayloadStruct\\s+(\\S+)\\s*=\\s*(\\S+)");
         for (IASTNode comment : comments) {
-            Matcher matcher1 = pattern1.matcher(comment.getRawSignature());
-            if (matcher1.find()) { // associate ID to struct
-                String canName = matcher1.group(1);
-                String structName = matcher1.group(2);
-                canNameToStruct.put(canName, structName);
+            Matcher matcher = pattern.matcher(comment.getRawSignature());
+            if (matcher.find()) { // associate ID to struct
+                String canName = matcher.group(1);
+                String structName = matcher.group(2);
+                decoderRepo.put(canName, decoderRepo.get(structName));
             }
         }
     }
 
+    /**
+     * Creates and stores a Primitive decoder if declaration is a Const
+     * @param declaration
+     * @throws Exception
+     */
```

Comment:
Does this actually throw an Exception?

I'm also not sure that `throws Exception` is too useful, if there's a specific exception this might throw (and the caller needs to be aware of) that might be more useful to document.